PaillierTests/PaillierEncryptionTests.cs

```diff
@@ -0,0 +1,169 @@
+
+                Paillier decryptAlgorithm = new PaillierManaged();
+                decryptAlgorithm.FromXmlString(algorithm.ToXmlString(true));
+
+                var z = new BigInteger(0);
+
+                var z_enc = encryptAlgorithm.EncryptData(z.getBytes());
+                var z_dec = decryptAlgorithm.DecryptData(z_enc);
+
+                var zero_array = new byte[z_dec.Length];
+                Array.Clear(zero_array, 0, zero_array.Length - 1);
+
+                CollectionAssert.AreEqual(zero_array, z_dec);
+            }
+        }
+
+        [TestMethod]
+        public void TestRandomBI()
+        {
+            // Failed test because of zeroes
+
+            Paillier algorithm = new PaillierManaged();
+            algorithm.Padding = PaillierPaddingMode.LeadingZeros;
+
+            for (algorithm.KeySize = 384; algorithm.KeySize <= 1088; algorithm.KeySize += 8)
+            {
+                Paillier encryptAlgorithm = new PaillierManaged();
+                encryptAlgorithm.FromXmlString(algorithm.ToXmlString(false));
+
+                Paillier decryptAlgorithm = new PaillierManaged();
+                decryptAlgorithm.FromXmlString(algorithm.ToXmlString(true));
+
+                var z = new BigInteger();
+                z.genRandomBits(new Random().Next(1, 2241), new RNGCryptoServiceProvider());
+
+                var z_enc = encryptAlgorithm.EncryptData(z.getBytes());
+                var z_dec = decryptAlgorithm.DecryptData(z_enc);
+
+                CollectionAssert.AreEqual(z.getBytes(), z_dec);
+            }
+        }
+
+        [TestMethod]
+        public void TestTextEncryption()
+        {
+            // Test failed somehow. 
+            // UPDATE: This test unexpectedly passed. I guess it's due to the BigInteger package update.
+            string message = "This is to test Paillier encryption and hopefully this message contains more than 2 blocks please please please please please please please please please please please pleaseplease please please pleaseplease please please please          ";
+            PaillierPaddingMode padding = PaillierPaddingMode.Zeros;
+
+            var plaintext = Encoding.Default.GetBytes(message);
+
+            Paillier algorithm = new PaillierManaged();
+
+            for (int keySize = 384; keySize <= 1088; keySize += 8)
+            {
+                algorithm.KeySize = keySize;
+                algorithm.Padding = padding;
+
+                Paillier encryptAlgorithm = new PaillierManaged();
+                encryptAlgorithm.FromXmlString(algorithm.ToXmlString(false));
+
+                var ciphertext = encryptAlgorithm.EncryptData(plaintext);
+
+                Paillier decryptAlgorithm = new PaillierManaged();
+                decryptAlgorithm.FromXmlString(algorithm.ToXmlString(true));
+
+                var candidatePlaintext = decryptAlgorithm.DecryptData(ciphertext);
+
+                var strip_zeros = StripTrailingZeros(candidatePlaintext, plaintext.Length);
```

Comment:
This is why text encryption works well. Actually, we don't have to need to strip trailing zeros. There's something with decrypting blocks padded with `PaillierPaddingMode.Zeros`. Remove this line and method `StripTrailingZeros` so that we later can figure why it doesn't work.